pkg/controller/pravegacluster/upgrade_test.go

```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
+			})
+		})
+		Context("syncSegmentStoreVersionTo07", func() {
+			var (
+				err          error
+				foundPravega *v1beta1.PravegaCluster
+				client       client.Client
+				sts, sts1    *appsv1.StatefulSet
+			)
+			BeforeEach(func() {
+				client = fake.NewFakeClient(p)
+				r = &ReconcilePravegaCluster{client: client, scheme: s}
+				_, _ = r.Reconcile(req)
+				foundPravega = &v1beta1.PravegaCluster{}
+				r.Reconcile(req)
+				sts = &appsv1.StatefulSet{}
+				p.Spec.Version = "0.7.0"
+				p.WithDefaults()
+				sts = pravega.MakeSegmentStoreStatefulSet(p)
+				client.Create(context.TODO(), sts)
+				_, _ = r.Reconcile(req)
+				_ = r.client.Get(context.TODO(), types.NamespacedName{Name: sts.Name, Namespace: p.Namespace}, sts)
+				_ = r.client.Get(context.TODO(), types.NamespacedName{Name: p.Name, Namespace: p.Namespace}, foundPravega)
+				sts.Status.ReadyReplicas = 5
+				*sts.Spec.Replicas = 5
+				r.client.Update(context.TODO(), sts)
+				foundPravega.Spec.Version = "0.5.0"
+				p.Spec.Version = "0.5.0"
+				p.WithDefaults()
+				sts1 = pravega.MakeSegmentStoreStatefulSet(p)
+				client.Create(context.TODO(), sts1)
+				sts1.Status.ReadyReplicas = 2
+				*sts1.Spec.Replicas = 2
+				r.client.Update(context.TODO(), sts1)
+				r.client.Update(context.TODO(), foundPravega)
+
+				_, err = r.syncSegmentStoreVersionTo07(foundPravega)
+				foundPravega.Status.TargetVersion = "0.5.0"
+
+			})
+			It("Error should be nil", func() {
+				Ω(err).Should(BeNil())
+			})
+
+		})
+
+		Context("checkUpdatedPods with faulty and non-faultypod", func() {
+			var (
+				result1, result2 bool
+				pod, pod1        []*corev1.Pod
+				err, err1        error
+			)
+			BeforeEach(func() {
+				testpod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "test"}, Spec: v1.PodSpec{Containers: []v1.Container{{Image: "testimage"}}},
+					Status: v1.PodStatus{
+						ContainerStatuses: []v1.ContainerStatus{
+							{
+								Name: "test",
+								State: v1.ContainerState{
+									Waiting: &v1.ContainerStateWaiting{
+										Reason: "CrashLoopBackOff",
+									},
+								},
+							},
+						}},
+				}
+				testpod1 := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "test"}, Spec: v1.PodSpec{Containers: []v1.Container{{Image: "testimage"}}},
+					Status: v1.PodStatus{
+						ContainerStatuses: []v1.ContainerStatus{
+							{
+								Name: "test1",
+							},
+						},
+					},
+				}
+				r.client.Create(context.TODO(), testpod)
+				r.client.Create(context.TODO(), testpod1)
+				r.client.Get(context.TODO(), types.NamespacedName{Name: "test", Namespace: "default"}, testpod)
+				r.client.Get(context.TODO(), types.NamespacedName{Name: "test1", Namespace: "default"}, testpod1)
+				pod = append(pod, testpod)
+				result1, err = r.checkUpdatedPods(pod, "0.7.1")
+				pod1 = append(pod1, testpod1)
+				result2, err1 = r.checkUpdatedPods(pod1, "0.7.1")
+			})
+			It("It should return false", func() {
```

Comment:
can we have different assertion for result1 and result2

Reply:
done